client/linuxabi/linux_abi.go

```diff
@@ -106,87 +112,185 @@ type SnpReportReq struct {
 	reserved [28]byte
 }
 
-// SnpReportResp is Linux's sev-guest ioctl abi for receiving a GET_REPORT response.
-type SnpReportResp struct {
+// SnpReportRespABI is Linux's sev-guest ioctl abi for receiving a GET_REPORT response.
+type SnpReportRespABI struct {
+	Status     uint32
+	ReportSize uint32
+	reserved   [0x20 - 8]byte
 	// Data is the response data, see SEV-SNP spec for the format
-	Data [abi.ReportSize]uint8
+	Data [SnpReportRespReportSize]uint8
 }
 
-// SnpExtendedReportReqSafe is close to Linux's sev-guest ioctl abi for sending a GET_EXTENDED_REPORT request,
-// but uses safer types for the Ioctl interface.
-type SnpExtendedReportReqSafe struct {
-	Data SnpReportReq
+// ABI returns the same object since it doesn't need a separate representation across the interface.
+func (r *SnpReportReqABI) ABI() BinaryConversion { return r }
+
+// Pointer returns a pointer to the object itself.
+func (r *SnpReportReqABI) Pointer() unsafe.Pointer {
+	return unsafe.Pointer(r)
+}
+
+// Finish is a no-op.
+func (r *SnpReportReqABI) Finish(b BinaryConvertible) error { return nil }
+
+// ABI returns the same object since it doesn't need a separate representation across the interface.
+func (r *SnpReportRespABI) ABI() BinaryConversion { return r }
+
+// Pointer returns a pointer to the object itself.
+func (r *SnpReportRespABI) Pointer() unsafe.Pointer {
+	return unsafe.Pointer(r)
+}
+
+// Finish checks the status of the message and translates it to a Golang error.
+func (r *SnpReportRespABI) Finish(b BinaryConvertible) error {
+	if r.Status != 0 {
+		switch r.Status {
+		case 0x16:
```

Comment:
Can you add a comment explaining this status and where it came from?

Reply:
Done.